### PR TITLE
Rewrite conversion stuff to use new pytave features

### PR DESCRIPTION
**Pull request by genuinelucifer**
Re-did the PR to `pytave` branch...

### CODE REVIEW

**Comment by genuinelucifer** [comment]
@cbm755 Please review this / merge if it's okay...

File: inst/private/check_and_convert.m [hunk 1]
```diff
@@ -1,63 +1,96 @@
...
-      pyexec(strjoin({'_allKeysStr = []',
-                      '_allValues = []',
-                      ['for key, value in ' cur_var '.iteritems():'],
-                      '    _allKeysStr.append(str(key))',
-                      '    _allValues.append(value)'}, newl));
-      _allKeysStr = pyeval('_allKeysStr');
+      obj{i} = check_and_convert(cur_pyobj);
+    elseif dict_to_struct
+      %if cur_var is dictionary with symbols/strings as keys then convert it to a struct
+      %TODO: will be better after https://bitbucket.org/mtmiller/pytave/issues/62  is resolved
```
Comment (cbm755) [inline inst/private/check_and_convert.m]: is this done now?

Comment (genuinelucifer) [inline inst/private/check_and_convert.m]: My laptop isn't working from couole of days, I have given it for repairs. I'll check this as soon as I get my laptop back. I think it should be working now as upstream has struct conversion now.

**Comment by cbm755** [comment]
Discussing performance of pycall here: https://bitbucket.org/mtmiller/pytave/issues/64/pycalls-are-too-slow

**Comment by cbm755** [comment]
my pytave branch wasn't working with master anyway, so merged there.  Maybe not quite ready for master yet.

**Pull request merged**